src/find_type_for_node.php

```diff
@@ -0,0 +1,78 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\__Private;
+;
+
+
+use namespace HH\Lib\Str;
+use namespace Facebook\TypeAssert;
+use namespace Facebook\HHAST;
+use type Facebook\HHAST\EditableNode;
+
+type TTypeAtPosOutput = shape(
+  ?'full_type' => shape(
+    'kind' => string,
+    ?'name' => string,
+    ...
+  ),
+  ...
+);
+
+function find_type_for_node(
+  EditableNode $root,
+  EditableNode $node,
+  string $path,
+): ?string {
+
+  list($line, $offset) = HHAST\find_position($root, $node);
+  $path = \realpath($path);
+  $command = vec[
+    'hh_client',
+    '--json',
+    '--from',
+    'hhast',
+    '--type-at-pos',
+    $line.':'.($offset + 1),
+    '<',
+    \escapeshellarg($path),
+  ];
+  $output = \tempnam(\sys_get_temp_dir(), 'hhast-temp');
+  \exec(Str\join($command, ' ').' >'.\escapeshellarg($output));
```

Comment:
make this function async, and use/extend `__Private\execute_async` if possible

Reply:
Resolved! I had to make some changes due to the escaping. It was not possible to do input redirection due to the escaping built into `execute_async`, and I was having trouble getting `hh_client` to not complain when passed the path to the file (`hh_client --type-at-pos /tmp/file:line:offset` would complain about hh_server being stopped).

Turns out it wanted a path to a directory as a second arg, adding that allowed me to pass the file path and the containing directory, skipping input redirection. i.e. i.e. `hh_client --type-at-pos /tmp/file:line:offset /tmp`